README.md

```diff
@@ -188,113 +270,104 @@ The following CSS:
 
+ - CSS variables may be declared in any normal selector, including many of the more complex ones.
+ - CSS variables may be in `@media`, `@support`, and other at-rules.
+ - `:hover` and other psuedo-classes are properly supported.
 
+Continue to the next section to see where some of these might be unsafe to do. There are reasons why `postcss-custom-properties` is very limited in what it supports, due to differing opinions on what is okay to support.
 
-# Syntax
+# Caveats
 
-### [Defining Custom Properties: the --* family of properties](http://dev.w3.org/csswg/css-variables/#defining-variables)
+When you declare a CSS variable inside one selector, but consume it in another, this does make unsafe assumptions which can be non-conforming in certain edge cases. Here is an example where these limitations end up with non-conforming behavior.
 
-A custom property is any property whose name starts with two dashes (U+002D HYPHEN-MINUS). A property must be in a `rule`.
+ - CSS:
 
-*Note: `:root` is nothing more than the selector for the root DOM node. You could use any other selector (class, id, etc)*
-```
-:root {
-	--foo-width: 100px;
-	--foo-bg-color: rgba(255, 0, 0, 0.85);
-}
-```
+   ```css
+   .component {
+     --text-color: blue;
+   }
 
-A custom property can be declared multiple times.
+   .component .header {
+     color: var(--text-color);
+   }
 
-*See: [W3C Draft: CSS Custom Properties for Cascading Variables, section 2](http://dev.w3.org/csswg/css-variables/#defining-variables) for more information.*
+   .component .text {
+     --text-color: green;
+     color: var(--text-color);
+   }
+   ```
+   
+ - Jade (for brevity) with resulting color:
 
+   ```jade
+   .component: Black text
+   .component: .header: Blue text
+   .component: .header: .text: Green text
+   .component: .header: .text: .header: Green text per spec, blue per this polyfill
+   ```
 
-### [Using Cascading Variables: the var() notation](http://dev.w3.org/csswg/css-variables/#using-variables)
+This spec deviation is intentional, as there's nothing this tool can do about that. `postcss-custom-properties` avoids this problem entirely by restricting itself to just the `:root` selector, and this is because the developers there would prefer to not support a feature instead of supporting it almost-correctly like what `postcss-css-variables` does.
 
+A related example in the JavaScript world is [Babel's](https://babeljs.io) transformation of [arrow functions](https://babeljs.io/docs/plugins/transform-es2015-arrow-functions/) and methods, where Babel does not prevent arrow functions or object methods from being called with `new`, nor does it modify the `prototype` property on them, for performance reasons.
```

Comment:
Remove, I don't this example is necessary to justify why this plugin should exist.